hamilton/io/utils.py

```diff
@@ -17,3 +24,28 @@ def get_file_metadata(path: str) -> Dict[str, Any]:
         "last_modified": os.path.getmtime(path),
         "timestamp": datetime.now().utcnow().timestamp(),
     }
+
+
+def get_sql_metadata(query_or_table: str, results: Union[int, pd.DataFrame]) -> Dict[str, Any]:
+    """Gives metadata from reading a SQL table or writing to SQL db.
+    This includes:
+    - the number of rows read or added
+    - the sql query (e.g., "SELECT foo FROM bar")
+    - the table name (e.g., "bar")
+    - the current time
+    """
+    query = query_or_table if "SELECT" in query_or_table else None
+    table_name = query_or_table if "SELECT" not in query_or_table else None
+    rows = (
+        results
+        if isinstance(results, int)
+        else len(results)
+        if isinstance(results, pd.DataFrame)
+        else None
```

Comment:
can still do a `len` on a dataframe though?

Reply:
the second if block calls len on the dataframe! the ternary expression looks confusing, so I switched to a regular if/elif block. it should read more clearly.